test/e2e/nfd_topology_updater.go

```diff
@@ -0,0 +1,119 @@
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	extclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	e2enetwork "k8s.io/kubernetes/test/e2e/framework/network"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+
+	testutils "sigs.k8s.io/node-feature-discovery/test/e2e/utils"
+)
+
+var _ = framework.KubeDescribe("[NFD] Node topology updater", func() {
+	f := framework.NewDefaultFramework("node-topology-updater")
+
+	ginkgo.Context("with single nfd-master pod", func() {
+		var (
+			extClient                *extclient.Clientset
+			crd                      *apiextensionsv1.CustomResourceDefinition
+			masterPod                *v1.Pod
+			masterService            *v1.Service
+			topologyUpdaterDaemonSet *appsv1.DaemonSet
+		)
+
+		ginkgo.BeforeEach(func() {
+			var err error
+
+			if extClient == nil {
+				extClient, err = extclient.NewForConfig(f.ClientConfig())
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+
+			ginkgo.By("Creating the node resource topologies CRD")
+			crd = testutils.NewNodeResourceTopologies()
+			_, err = extClient.ApiextensionsV1().CustomResourceDefinitions().Create(context.TODO(), crd, metav1.CreateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			err = testutils.ConfigureRBAC(f.ClientSet, f.Namespace.Name)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			image := fmt.Sprintf("%s:%s", *dockerRepo, *dockerTag)
+			masterPod = f.PodClient().CreateSync(testutils.NFDMasterPod(image, false))
+
+			// Create nfd-master service
+			masterService, err = testutils.CreateService(f.ClientSet, f.Namespace.Name)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Waiting for the nfd-master service to be up")
+			gomega.Expect(e2enetwork.WaitForService(f.ClientSet, f.Namespace.Name, masterService.Name, true, time.Second, 10*time.Second)).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Creating nfd-topology-updater daemonset")
+			topologyUpdaterDaemonSet = testutils.NFDTopologyUpdaterDaemonSet(fmt.Sprintf("%s:%s", *dockerRepo, *dockerTag), []string{})
+			topologyUpdaterDaemonSet, err = f.ClientSet.AppsV1().DaemonSets(f.Namespace.Name).Create(context.TODO(), topologyUpdaterDaemonSet, metav1.CreateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Waiting for daemonset pods to be ready")
+			gomega.Expect(e2epod.WaitForPodsReady(f.ClientSet, f.Namespace.Name, topologyUpdaterDaemonSet.Spec.Template.Labels["name"], 5)).NotTo(gomega.HaveOccurred())
+		})
+
+		ginkgo.FIt("should fill the node resource topologies CR with the data", func() {
+
```

Comment:
uhm, not sure if adding everything in one go is actually clearer.
I don't have any strong opinion at all, just wondering

Reply:
IMHO it will save us some time during setup and teardown. and I think other tests will be something like, start the container that requests different resources and check that it reflected under the node resource topologies CR. So IMHO for the basic test to put all under the same `It` should be good enough.